docs/internal/vendored-code.md

```diff
@@ -0,0 +1,38 @@
+# Vendored code
+
+To avoid a direct dependency on external packages, some code is vendored.
+
+This means that source code from external repositories is copied into the `/src/OpenTelemetry.AutoInstrumentation/Vendors/{LibName}`
+folder.
+
+To make vendored code fully internal, apply the following changes:
+
+1. Prefix all namespaces with `Vendors.` (e.g., change `YamlDotNet` to `Vendors.YamlDotNet`).
+2. Update all using directives to reference the vendored namespaces
+   (e.g., change `using YamlDotNet` to `using Vendors.YamlDotNet`).
+3. Mark whole public contract as internal.
+4. To suppress static analysis and enable nullable reference types,
```

Comment:
Is it possible to add a Directory.Build.props file at this vendored folder level to help simplify making changes across multiple files?